firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -320,7 +324,8 @@ public static Object convertToPlainJavaTypes(Object object) {
     } else if (Character.class.isAssignableFrom(clazz) || char.class.isAssignableFrom(clazz)) {
       throw deserializeError(path, "Deserializing to chars is not supported");
     } else {
-      throw new IllegalArgumentException("Unknown primitive type: " + clazz);
+      throw deserializeError(
```

Comment:
Similarly, do we need to special case `short`, `byte`, and `char` if we give a nearly equivalent message here?

Reply:
We can probably reduce our code duplication here as well. I removed the extra checks.